.github/workflows/web-tag.yml

```diff
@@ -16,6 +16,29 @@ on:
       - "v[0-9]+.[0-9]+.[0-9]+"
 
 jobs:
+  generateChangelog:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-node@v4
+        with:
+          node-version: 20
+          cache: 'npm'
+          cache-dependency-path: app/package-lock.json
+      - name: Install dependencies
+        run: npm ci --prefix app
+      - name: Generate changelog
+        env:
+          OPENAI_API_KEY: ${{ secrets.OPENAI_API_KEY_PROD }}
+        run: npm run generate:changelog --prefix app
+      - name: Commit and push changelog
+        run: |
+          git config --global user.name 'github-actions[bot]'
+          git config --global user.email 'github-actions[bot]@users.noreply.github.com'
+          git add CHANGELOG.md
+          git diff --cached --quiet || git commit -m 'chore: update CHANGELOG.md [skip ci]'
+          git push
```

Comment:
I had some problems doing a `git push` from a GitHub action previously on `develop`, but this might work on `main`.